### ISSUE TITLE
Issues opening ClamAV database

### ISSUE THREAD

**Issue by RicochetYT** [kpcyrd/libredefender#3]
Hello,
I stumbled across this and decided that I'd give it a go, why not, looking for some antivirus stuff for Linux for the hell of it.
However, every time I want to do a scan, this occurs.
```[ricochet@gamer ~]$ libredefender scan
[2021-06-03T08:02:37Z INFO  libredefender::scan] Initializing with libclamav 0.103.2
[2021-06-03T08:02:37Z WARN  libredefender::db] Failed to open existing database, using new one: Failed to open database: No such file or directory (os error 2)
[2021-06-03T08:02:37Z INFO  libredefender::scan] Empty arguments, defaulting to home directory
[2021-06-03T08:02:37Z INFO  libredefender::scan] Loading database from /var/lib/clamav...
[2021-06-03T08:02:45Z INFO  libredefender::scan] Checking database age...
Error: Failed to open clamav database

Caused by:
    No such file or directory (os error 2)
[ricochet@gamer ~]$

// However, I already have ClamAV setup and I have ran freshclam before.

// Here is a listing of my files in /var/lib/clamav:

[ricochet@gamer ~]$ cd /var/lib/clamav
[ricochet@gamer clamav]$ ls
bytecode.cvd  daily.cvd  main.cvd  mirrors.dat

**Comment by kpcyrd** [comment]
@RicochetYT thanks for the report!

Can you try again with the current main? I'm going to push a new release if it's working for you. :)

**Comment by RicochetYT** [comment]
> @RicochetYT thanks for the report!
> 
> Can you try again with the current main? I'm going to push a new release if it's working for you. :)

Gladly!

**Comment by RicochetYT** [comment]
> @RicochetYT thanks for the report!
> 
> Can you try again with the current main? I'm going to push a new release if it's working for you. :)

Gives me a warn for a failure but seems to work(?) anyways.
Compiled from the latest commit @ https://github.com/kpcyrd/libredefender/commit/1c019febb6e6700e2f6c36eddc03f67ce2ed50ca

```[ricochet@gamer bin]$ ./libredefender -V
libredefender 0.2.0
[ricochet@gamer bin]$ ./libredefender --verbose scan
[2021-06-03T23:36:29Z DEBUG libredefender::nice] Calling ioprio_set for idle priority
[2021-06-03T23:36:29Z DEBUG libredefender::nice] Calling nice(2) for idle priority
[2021-06-03T23:36:29Z INFO  libredefender::scan] Initializing with libclamav 0.103.2
[2021-06-03T23:36:29Z WARN  libredefender::db] Failed to open existing database, using new one: Failed to open database: No such file or directory (os error 2)
[2021-06-03T23:36:29Z INFO  libredefender::scan] Scanning home directory: "/home/ricochet"
[2021-06-03T23:36:29Z INFO  libredefender::scan] Loading database from /var/lib/clamav...
[2021-06-03T23:36:35Z INFO  libredefender::scan] Checking database age...
[2021-06-03T23:36:35Z DEBUG libredefender::scan] Checking if database exists: "/var/lib/clamav/daily.cld"
[2021-06-03T23:36:35Z INFO  libredefender::scan] Compiling clamav rules...
[2021-06-03T23:36:37Z INFO  libredefender::scan] Spawning 12 scanner(s)...
[2021-06-03T23:36:37Z INFO  libredefender::scan] Scanning directory /home/ricochet...

// Current /var/lib/clamav file structure
[ricochet@gamer clamav]$ ls
bytecode.cvd  daily.cld  main.cvd  mirrors.dat

**Comment by kpcyrd** [comment]
The warning is fine and only happens during the first run, I've fixed this in #5 regardless though.

I did some research and apparently freshclam writes into `daily.cvd` during the first download and starting from the second download on the file is called `daily.cld`. You've downloaded an update in the meantime, so the original binary you've used would likely work now as well. :)
